BlazarService/src/main/java/com/hubspot/blazar/listener/LaunchingRepositoryBuildVisitor.java

```diff
@@ -73,7 +75,16 @@ protected void visitLaunching(RepositoryBuild build) throws Exception {
 
     final Set<Module> toBuild = new HashSet<>();
     if (build.getBuildTrigger().getType() == Type.MANUAL) {
-      toBuild.addAll(modules);
+      if (build.getBuildTrigger().getModuleIds().isEmpty()) {
+        toBuild.addAll(modules);
+      } else {
+        final Set<Integer> moduleIds = build.getBuildTrigger().getModuleIds();
+        for (Module module : modules) {
+          if (moduleIds.contains(module.getId().or(-1))) {
+            toBuild.add(module);
```

Comment:
This won't build downstream modules of the selected ones (ie, I select `OrionData` it won't build `OrionService` as well). Maybe whether to include downstreams should be an additional option in the modal window the user gets? At that point maybe we should move these fields from the `BuildTrigger` object into a new `BuildOptions` object because @also wanted some options as well (enable SSH access, clear the cache directories, etc.) which could get added to this modal and to the `BuildOptions` object.

This is a more involved change (new DB column, new field on the RepositoryBuild) but I think it might be better in the long run